src/main/java/org/broadinstitute/hellbender/engine/spark/AddContextDataToReadSpark.java

```diff
@@ -45,4 +63,53 @@
+            final JavaRDD<GATKRead> reads, final ReferenceMultiSource referenceSource,
+            final JavaRDD<GATKVariant> variants, final SAMSequenceDictionary sequenceDictionary) {
+        JavaRDD<GATKRead> mappedReads = reads.filter(read -> ReadFilterLibrary.MAPPED.test(read));
+
+        final List<SimpleInterval> intervals = IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+        // use unpadded shards (padding is only needed for reference bases)
+        final List<ShardBoundary> intervalShards = intervals.stream()
+                .flatMap(interval -> Shard.divideIntervalIntoShards(interval, SHARD_SIZE, SHARD_SIZE, 0, sequenceDictionary).stream())
+                .collect(Collectors.toList());
+
+        final Broadcast<ReferenceMultiSource> bReferenceSource = ctx.broadcast(referenceSource);
+        final IntervalsSkipList<GATKVariant> variantSkipList = new IntervalsSkipList<>(variants.collect());
+        final Broadcast<IntervalsSkipList<GATKVariant>> variantsBroadcast = ctx.broadcast(variantSkipList);
+
+        JavaRDD<Shard<GATKRead>> shardedReads = SparkUtils.shard(ctx, mappedReads, GATKRead.class, sequenceDictionary, intervalShards);
+        return shardedReads.flatMapToPair(new PairFlatMapFunction<Shard<GATKRead>, GATKRead, ReadContextData>() {
+            private static final long serialVersionUID = 1L;
+            @Override
+            public Iterable<Tuple2<GATKRead, ReadContextData>> call(Shard<GATKRead> shard) throws Exception {
+                // get reference bases for this shard (padded)
+                SimpleInterval paddedInterval = shard.getInterval().expandWithinContig(SHARD_PADDING, sequenceDictionary);
+                ReferenceBases referenceBases = bReferenceSource.getValue().getReferenceBases(null, paddedInterval);
+                final IntervalsSkipList<GATKVariant> intervalsSkipList = variantsBroadcast.getValue();
+                Iterator<Tuple2<GATKRead, ReadContextData>> transform = Iterators.transform(shard.iterator(), new Function<GATKRead, Tuple2<GATKRead, ReadContextData>>() {
+                    @Nullable
+                    @Override
+                    public Tuple2<GATKRead, ReadContextData> apply(@Nullable GATKRead r) {
+                        List<GATKVariant> overlappingVariants;
+                        if (SimpleInterval.isValid(r.getContig(), r.getStart(), r.getEnd())) {
+                            overlappingVariants = intervalsSkipList.getOverlapping(new SimpleInterval(r));
+                        } else {
+                            //Sometimes we have reads that do not form valid intervals (reads that do not consume any ref bases, eg CIGAR 61S90I
+                            //In those cases, we'll just say that nothing overlaps the read
+                            overlappingVariants = Collections.emptyList();
+                        }
+                        return new Tuple2<>(r, new ReadContextData(referenceBases, overlappingVariants));
```

Comment:
No, it's not used, but the padding will likely give enough context and is a more efficient way of doing this. Address as a follow on?